documentation/developer-guide/modules/tooling-guide/pages/maven-plugin.adoc

```diff
@@ -0,0 +1,424 @@
+
+=== Generating HTML Documentation
+
+The `generateDocumentation` goal generates HTML reference documentation for a given Aspect Model. The default life cycle phase for the goal is `generate-resources`.
+
+Usage:
+
+[source,maven,subs=attributes+]
+----
+<build>
+  <plugins>
+     <plugin>
+        <artifactId>sds-aspect-model-maven-plugin</artifactId>
+        <executions>
+           <execution>
+              <id>generate-html-doc</id>
+              <goals>
+                 <goal>generateDocumentation</goal>
+              </goals>
+           </execution>
+        </executions>
+        <configuration>
+           <aspectModelFilePath>{path-to-aspect-model/AspectModelFile.ttl}</aspectModelFilePath>
+           <outputDirectory>{directory-for-generated-source-files}</outputDirectory>
+        </configuration>
+     </plugin>
+  </plugins>
+</build>
+----
+
+Configuration Properties:
+
+[width="100%", options="header", cols="20,50,10,10,10"]
+|===
+| Property | Description | Type | Default Value | Required
+| `aspectModelFilePath` | The path to the Aspect Model file. | `String` | none | {nok}
+| `outputDirectory` | The path to the directory where the generated HTML document will be written to. | `String` | none | {ok}
+| `htmlCustomCSSFilePath` | Path to a CSS file with custom styles to be included in the generated HTML documentation. | `String` | none | {nok}
+|===
+
+=== Generating SVG or PNG Diagrams
+
+The `generateDiagram` goal generates an automatically layouted diagram for a given Aspect Model in SVG, PNG or https://www.graphviz.org/[Graphviz/DOT]. The default life cycle phase for the goal is `generate-resources`.
+
+Usage:
+
+[source,maven,subs=attributes+]
+----
+<build>
+  <plugins>
+     <plugin>
+        <artifactId>sds-aspect-model-maven-plugin</artifactId>
+        <executions>
+           <execution>
+              <id>generate-aspect-model-diagram</id>
+              <goals>
+                 <goal>generateDiagram</goal>
+              </goals>
+           </execution>
+        </executions>
+        <configuration>
+           <aspectModelFilePath>{path-to-aspect-model/AspectModelFile.ttl}</aspectModelFilePath>
+           <outputDirectory>{directory-for-generated-source-files}</outputDirectory>
+           <targetFormats>
+              <targetFormat>png</targetFormat>
+           </targetFormats>
+        </configuration>
+     </plugin>
+  </plugins>
+</build>
+----
+
+Configuration Properties:
+
+[width="100%", options="header", cols="20,50,10,10,10"]
+|===
+| Property | Description | Type | Default Value | Required
+| `aspectModelFilePath` | The path to the Aspect Model file. | `String` | none | {nok}
+| `outputDirectory` | The path to the directory where the generated diagrams will be written to. | `String` | none | {ok}
+| `targetFormats` | A list formats in which the diagram(s) will be created. A diagram will be generated for each specified format. | `String` | none | {ok}
```

Comment:
Should mention the valid formats

Reply:
The valid formats are mentioned in the description above. Would additionally want to include them here?

Reply:
This is fine then.